addTwonumbers.py

```diff
@@ -0,0 +1,74 @@
+#https://leetcode.com/problems/add-two-numbers/
+#You are given two non-empty linked lists representing two non-negative integers. The digits are stored in reverse order and each of their nodes contain a single digit. Add the two numbers and return it as a linked list.
+#You may assume the two numbers do not contain any leading zero, except the number 0 itself.
+#Example:
+#Input: (2 -> 4 -> 3) + (5 -> 6 -> 4)
+#Output: 7 -> 0 -> 8
+#Explanation: 342 + 465 = 807.
+
+ #Definition for singly-linked list.
+
+
+class ListNode:
+ def __init__(self, x):
+     self.val = x
+     self.next = None
+
+
+class Solution:
+    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
+        _result = node = ListNode(0)
+        carry = 0
+        while l1 or l2:
+            v1 = l1.val if l1 else 0
+            v2 = l2.val if l2 else 0
+            node.next = ListNode((v1 + v2 + carry) % 10)
+            node = node.next
+            carry = (v1 + v2 + carry) // 10
+            l1 = l1.next if l1 else None
+            l2 = l2.next if l2 else None
+        if carry > 0:
+            node.next = ListNode(carry)
+        return _result.next
+
+    def printlist(self, l: ListNode):
```

Comment:
nit return type ` -> List[int]` see Mypy

Reply:
?

Reply:
return self.printlist(_result.next) like this?

Reply:
forget it